test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -801,67 +801,135 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
 
     #endregion Redirect tests
 
-    #region SkipHeaderVerification Tests
+    Context "Invoke-WebRequest SkipHeaderVerification Tests" {
+        BeforeAll {
+            $Testfile = Join-Path $testdrive 'SkipHeaderVerification.txt'
+            'bar' | Set-Content $Testfile
+        }
 
-    It "Verifies Invoke-WebRequest default header handling with no errors" {
-        $uri = Get-WebListenerUrl -Test 'Get'
-        $headers = @{"If-Match" = "*"}
-        $response = ExecuteRequestWithCustomHeaders -Uri $uri -headers $headers
+        It "Verifies Invoke-WebRequest default header handling with no errors" {
+            $uri = Get-WebListenerUrl -Test 'Get'
+            $headers = @{"If-Match" = "*"}
+            $response = ExecuteRequestWithCustomHeaders -Uri $uri -headers $headers
 
-        $response.Error | Should BeNullOrEmpty
-        $response.Content.Headers."If-Match" | Should BeExactly "*"
-    }
+            $response.Error | Should BeNullOrEmpty
+            $response.Content.Headers."If-Match" | Should BeExactly "*"
+        }
 
-    It "Verifies Invoke-WebRequest default header handling reports an error is returned for an invalid If-Match header value" {
-        $uri = Get-WebListenerUrl -Test 'Get'
-        $headers = @{"If-Match" = "12345"}
-        $response = ExecuteRequestWithCustomHeaders -Uri $uri -headers $headers
+        It "Verifies Invoke-WebRequest default header handling reports an error is returned for an invalid If-Match header value" {
+            $uri = Get-WebListenerUrl -Test 'Get'
+            $headers = @{"If-Match" = "12345"}
+            $response = ExecuteRequestWithCustomHeaders -Uri $uri -headers $headers
 
-        $response.Error | Should Not BeNullOrEmpty
-        $response.Error.FullyQualifiedErrorId | Should Be "System.FormatException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
-        $response.Error.Exception.Message | Should Be "The format of value '12345' is invalid."
-    }
+            $response.Error | Should Not BeNullOrEmpty
+            $response.Error.FullyQualifiedErrorId | Should Be "System.FormatException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
```

Comment:
@dantraMSFT This is still WIP

Reply:
Understood.

Reply:
Actually... we do hit the `MediaTypeHeaderValue` try block already in the tests "Verifies Invoke-WebRequest default ContentType handling reports an error is returned for an invalid Content-Type header value and -Body" and "Verifies Invoke-RestMethod default ContentType handling reports an error is returned for an invalid Content-Type header value and -Body".

We actually can't reach the `CharSet` section (for now) because... `new MediaTypeHeaderValue("application/json; charset=utf-8");` will throw `FormatException`....  dotnet/corefx#16290 

Effectively... This is all dead code already:

https://github.com/PowerShell/PowerShell/blob/45bf1dc142f811fd98816ff749d3c03e8313fb71/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs#L1480-L1491

That would explain why I can't find any tests for `-ContentType 'text/plain; charset=utf-8'` or similar.. it wouldn't even be possible. It already throws a `FormatException`

Reply:
Closed